test/functional/rpc_fundrawtransaction.py

```diff
@@ -177,6 +178,38 @@ def test_simple(self):
         dec_tx  = self.nodes[2].decoderawtransaction(rawtxfund['hex'])
         assert len(dec_tx['vin']) > 0  #test that we have enough inputs
 
+    def check_witness_inputs(self, vins):
+        for vin in vins:
+            # check vin is a segwit input
+            utxo = self.nodes[2].gettxout(vin['txid'], vin['vout'])
+            info = self.nodes[2].getaddressinfo(utxo['scriptPubKey']['address'])
+            if not info['iswitness']:
+                return False
+
+        return True
+
+    def test_witness_only(self):
+        self.log.info("Testing fundrawtxn with witness inputs only")
+
+        inputs  = [ ]
+
+        # make sure legacy inputs are not accepted in witness only mode if no witness inputs are found
+        self.generatetoaddress(self.nodes[2], 2, self.nodes[2].getnewaddress(address_type='legacy'))
```

Comment:
This may actually be testing something else than intended: given that the coinbase output is not 100 blocks old yet, I suspect the "Insufficient funds" would be foremost caused by the inputs being immature.

This test could be improved by creating a scenario in which a transaction is first successfully funded by using legacy inputs, but then toggling the `'segwit_inputs_only': true` causes a repeated attempt to fund the transaction to fail.